src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -447,30 +447,60 @@ def create_workflow(
     )
 
 
-def create_detector(
+def get_detector_field_values(
     alert_rule: AlertRule,
-    project_id: int,
     data_condition_group: DataConditionGroup,
+    project_id: int | None = None,
     user: RpcUser | None = None,
-) -> Detector:
-    return Detector.objects.create(
-        project_id=project_id,
-        enabled=True,
-        created_by_id=user.id if user else None,
-        name=alert_rule.name,
-        workflow_condition_group=data_condition_group,
-        type=MetricAlertFire.slug,
-        description=alert_rule.description,
-        owner_user_id=alert_rule.user_id,
-        owner_team=alert_rule.team,
-        config={
+) -> dict[str, Any]:
+    detector_field_values = {
```

Comment:
can we make a typed dict or a data class that represents this? We could then use that in the method signature as the result type, so other methods using this can get a better sense of the shape of the data as it's passed around.